gtsfm/utils/metrics.py

```diff
@@ -167,19 +167,71 @@ def compute_averaging_metrics(
+    # ground truth is the reference/target for alignment. discard 2nd return arg -- the estimated Similarity(3) object
+    wTi_aligned_list, _ = comp_utils.align_poses_sim3_ignore_missing(gt_wTi_list, wTi_list)
+
+    i2Ui1_dict_gt = get_twoview_translation_directions(gt_wTi_list)
 
     wRi_aligned_list, wti_aligned_list = get_rotations_translations_from_poses(wTi_aligned_list)
     gt_wRi_list, gt_wti_list = get_rotations_translations_from_poses(gt_wTi_list)
 
     metrics = []
     metrics.append(compute_rotation_angle_metric(wRi_aligned_list, gt_wRi_list))
     metrics.append(compute_translation_distance_metric(wti_aligned_list, gt_wti_list))
-    metrics.append(compute_translation_angle_metric(i2Ui1_dict, wTi_aligned_list))
+    metrics.append(compute_translation_angle_metric(i2Ui1_dict=i2Ui1_dict_gt, wTi_list=wTi_aligned_list))
     return GtsfmMetricsGroup(name="averaging_metrics", metrics=metrics)
 
 
+def compute_ba_pose_metrics(
+    gt_wTi_list: List[Pose3],
+    ba_output: GtsfmData,
+) -> GtsfmMetricsGroup:
+    """Compute pose errors w.r.t. GT for the bundle adjustment result.
+
+    Note: inputs must be aligned beforehand to the ground truth.
+
+    Args:
+        gt_wTi_list: List of ground truth poses.
+        ba_output: sparse multi-view result, as output of bundle adjustment.
+
+    Returns:
+        A group of metrics that describe errors associated with a bundle adjustment result (w.r.t. GT).
+    """
+    wTi_aligned_list = ba_output.get_camera_poses()
+    i2Ui1_dict_gt = get_twoview_translation_directions(gt_wTi_list)
+
+    wRi_aligned_list, wti_aligned_list = get_rotations_translations_from_poses(wTi_aligned_list)
+    gt_wRi_list, gt_wti_list = get_rotations_translations_from_poses(gt_wTi_list)
+
+    metrics = []
+    metrics.append(compute_rotation_angle_metric(wRi_aligned_list, gt_wRi_list))
+    metrics.append(compute_translation_distance_metric(wti_aligned_list, gt_wti_list))
+    metrics.append(compute_translation_angle_metric(i2Ui1_dict_gt, wTi_aligned_list))
+    return GtsfmMetricsGroup(name="mvs_input_metrics", metrics=metrics)
+
+
+def get_twoview_translation_directions(wTi_list: List[Pose3]) -> Dict[Tuple[int, int], Unit3]:
+    """Generate synthetic measurements of the 2-view translation directions between image pairs.
+
+    Args:
+        wTi_list: List of poses (e.g. could be ground truth).
+
+    Returns:
+        i2Ui1_dict: Dict from (i1, i2) to unit translation direction i2Ui1.
+    """
+    number_images = len(wTi_list)  # vs. using ba_output.number_images()
+
+    # check against all possible image pairs -- compute unit translation directions
+    i2Ui1_dict = {}
+    possible_img_pair_idxs = list(itertools.combinations(range(number_images), 2))
+    for (i1, i2) in possible_img_pair_idxs:
+        # compute the exact relative pose
+        i2Ti1 = wTi_list[i2].between(wTi_list[i1])
+        i2Ui1_dict[(i1, i2)] = Unit3(i2Ti1.translation())
+
+    return i2Ui1_dict
+
```

Comment:
this shouldnt be in metrics.py

Reply:
@ayushbaid not sure where is a better place to put it though?

Reply:
Maybe `utils/pose`